proxy: pass connection options from agent

jmcshane opened the pull request:
This change allows the cilium-agent to specify key connection options to deal with connection reuse issues. The motivation for this change comes from connection issues from cilium-envoy to S3. As seen in this article https://forums.aws.amazon.com/thread.jspa?threadID=110998, S3 will reset a connection that is re-used after 100 requests. Envoy can address this by creating a new connection after that number of requests, but we need to pass that configuration from cilium-agent to cilium-envoy.

Signed-off-by: jmcshane <james.mcshane@superorbital.io>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

<!-- Description of change -->

Fixes: #19136

```release-note
proxy: Add proxy common http options arguments to agent
```

jrajahalme reviewed (approved):
Looks good, thanks!

sayboras reviewed (approved):
LGTM 💯

aanm commented on pkg/option/config.go:
```diff
@@ -163,6 +163,12 @@ const (
+	ProxyMaxRequestsPerConnection = "proxy-max-requests-per-connection"
+
+	// ProxyMaxConnectionDuration specifies the max_connection_duration setting for the proxy in seconds
+	ProxyMaxConnectionDuration = "proxy-max-connection-duration"
```
`proxy-max-connection-duration` -> `proxy-max-connection-duration-seconds`

aanm reviewed (changes requested):
Just one small nit. Thank you!

Commit:
proxy: pass connection options from agent

This change allows the cilium-agent to specify key connection options to deal with connection reuse issues. The motivation for this change comes from connection issues from cilium-envoy to S3. As seen in this article https://forums.aws.amazon.com/thread.jspa?threadID=110998, S3 will reset a connection that is re-used after 100 requests. Envoy can address this by creating a new connection after that number of requests, but we need to pass that configuration from cilium-agent to cilium-envoy.

Signed-off-by: jmcshane <james.mcshane@superorbital.io>

ldelossa reviewed (approved):
LGTM :+1:
